### PR TITLE
[AVFoundation] Address all the XAMCORE_4_0 notes in AVFoundation for .NET.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This PR is probably best reviewed commit-by-commit, since it contains many small and discrete commits.

### CODE REVIEW

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* introspection/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2414
* trimmode link/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2515

**Review by chamons — APPROVED** [review]
❤️  it. Commit by commit was a great story.

File: tests/xtro-sharpie/api-annotations-dotnet/MacCatalyst-AVFoundation.ignore [hunk 1]
```diff
@@ -28,3 +28,9 @@
 !missing-selector! AVSampleCursor::presentationTimeStamp not bound
 !missing-selector! AVSampleCursor::samplesWithEarlierDecodeTimeStampsMayHaveLaterPresentationTimeStampsThanCursor: not bound
 !missing-selector! AVSampleCursor::samplesWithLaterDecodeTimeStampsMayHaveEarlierPresentationTimeStampsThanCursor: not bound
+
+## deprecated in all Mac Catalyst versions we support, and there's a viable alternative to use instead.
+!missing-protocol-member! AVAudioRecorderDelegate::audioRecorderEndInterruption:withFlags: not found
+
+## this API is to have cute Objective-C syntax for something that's (according to the docs) identical to what another method do (which we've bound).
```
Comment (tj-devel709) [inline tests/xtro-sharpie/api-annotations-dotnet/MacCatalyst-AVFoundation.ignore:35]:
```suggestion
## this API is to have cute Objective-C syntax for something that's (according to the docs) identical to what another method does (which we've bound).
```

File: tests/xtro-sharpie/api-annotations-dotnet/iOS-AVFoundation.ignore [hunk 2]
```diff
@@ -3,3 +3,9 @@
 
 ## https://github.com/xamarin/xamarin-macios/issues/3213 should be fixed before conformance to 'AVQueuedSampleBufferRendering' is restored.
 !missing-protocol-conformance! AVSampleBufferDisplayLayer should conform to AVQueuedSampleBufferRendering (defined in 'AVSampleBufferDisplayLayerQueueManagement' category)
+
+## deprecated in all iOS versions we support, and there's a viable alternative to use instead.
+!missing-protocol-member! AVAudioRecorderDelegate::audioRecorderEndInterruption:withFlags: not found
+
+## this API is to have cute Objective-C syntax for something that's (according to the docs) identical to what another method do (which we've bound).
```
Comment (tj-devel709) [inline tests/xtro-sharpie/api-annotations-dotnet/iOS-AVFoundation.ignore:10]:
```suggestion
## this API is to have cute Objective-C syntax for something that's (according to the docs) identical to what another method does (which we've bound).
```

**Review by tj-devel709 — APPROVED** [review]
Just left a little tiny comment about a comment!

**Comment by rolfbjarne** [comment]
Test failures are unrelated (https://github.com/xamarin/maccore/issues/2443).

**Pull request merged**